<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GeneratePrimitiveDialog</class>
 <widget class="QDialog" name="GeneratePrimitiveDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>446</width>
    <height>193</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Generate Primitive</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QGroupBox" name="groupbox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Type</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="1" column="0">
           <widget class="QRadioButton" name="typeCircle">
            <property name="text">
             <string>Circle</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <property name="autoRepeat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QRadioButton" name="typeCylinder">
            <property name="text">
             <string>C&amp;ylinder</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QRadioButton" name="typeDisc">
            <property name="text">
             <string>Disc</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QRadioButton" name="typeDiscNegative">
            <property name="text">
             <string>Disc &amp;Negative</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QRadioButton" name="typeRing">
            <property name="text">
             <string>Ring</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QRadioButton" name="typeCone">
            <property name="text">
             <string>Cone</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="2" column="1">
          <widget class="QSpinBox" name="ringNumber">
           <property name="enabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Ring number:</string>
           </property>
           <property name="buddy">
            <cstring>ringNumber</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="segments">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>16</number>
           </property>
           <property name="value">
            <number>16</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Segments:</string>
           </property>
           <property name="buddy">
            <cstring>segments</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="divisions">
           <property name="currentIndex">
            <number>1</number>
           </property>
           <item>
            <property name="text">
             <string>8</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>16</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>48</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Di&amp;visions:</string>
           </property>
           <property name="buddy">
            <cstring>divisions</cstring>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>GeneratePrimitiveDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>263</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>GeneratePrimitiveDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>331</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>typeCircle</sender>
   <signal>clicked(bool)</signal>
   <receiver>ringNumber</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>45</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>433</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>typeCylinder</sender>
   <signal>clicked(bool)</signal>
   <receiver>ringNumber</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>433</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>typeDisc</sender>
   <signal>clicked(bool)</signal>
   <receiver>ringNumber</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>77</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>433</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>typeDiscNegative</sender>
   <signal>clicked(bool)</signal>
   <receiver>ringNumber</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>433</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>typeRing</sender>
   <signal>clicked(bool)</signal>
   <receiver>ringNumber</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>73</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>433</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>typeCone</sender>
   <signal>clicked(bool)</signal>
   <receiver>ringNumber</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>433</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>enableRingNumber()</slot>
 </slots>
</ui>
